src/main/scala/edu/ie3/simona/model/participant/StorageModel.scala

```diff
@@ -0,0 +1,378 @@
+/*
+ * © 2022. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.model.participant
+
+import edu.ie3.datamodel.models.input.system.StorageInput
+import edu.ie3.simona.agent.participant.data.Data.PrimaryData.ApparentPower
+import edu.ie3.simona.model.SystemComponent
+import edu.ie3.simona.model.participant.StorageModel.{
+  RefTargetSocParams,
+  StorageRelevantData,
+  StorageState,
+}
+import edu.ie3.simona.model.participant.control.QControl
+import edu.ie3.simona.ontology.messages.flex.FlexibilityMessage.ProvideFlexOptions
+import edu.ie3.simona.ontology.messages.flex.MinMaxFlexibilityMessage.ProvideMinMaxFlexOptions
+import edu.ie3.util.quantities.PowerSystemUnits
+import edu.ie3.util.scala.OperationInterval
+import edu.ie3.util.scala.quantities.DefaultQuantities._
+import squants.energy.{KilowattHours, Kilowatts}
+import squants.{Dimensionless, Each, Energy, Power, Seconds}
+
+import java.time.ZonedDateTime
+import java.util.UUID
+
+final case class StorageModel(
+    uuid: UUID,
+    id: String,
+    operationInterval: OperationInterval,
+    qControl: QControl,
+    sRated: Power,
+    cosPhiRated: Double,
+    eStorage: Energy,
+    pMax: Power,
+    eta: Dimensionless,
+    dod: Dimensionless,
+    initialSoc: Double,
+    targetSoc: Option[Double],
+) extends SystemParticipant[StorageRelevantData, ApparentPower, StorageState](
+      uuid,
+      id,
+      operationInterval,
+      qControl,
+      sRated,
+      cosPhiRated,
+    ) {
+
+  private val minEnergy = zeroKWH
+
+  /** Tolerance for power comparisons. Amounts to 1 W for 1 GWh storage
+    */
+  private implicit val powerTolerance: Power = eStorage / Seconds(1) / 3.6e12
```

Comment:
tl;dr I found out why we want it this way when I typed this out

I have some food for thought here myself. Adapting the powerTolerance to be relative to the capacity has been a todo for some time and has now been solved here. For reference, this value is currently only used when checking whether the charging power is close to zero, which is only checked once here (`setPower ~= zeroKW`).

So now I have some questions. Does it really make sense to base the power tolerance on capacity, and the `toleranceMargin` related to SOC based on the maximum power? I feel like both should be rather calculated from `pMax`, because I think it's not impossible to have a battery with large storage and rather small charging power and then we should base comparisons of power values on the maximum power, not capacity.

I think the problem that we're trying to solve here is that in a situation where we charge with a very small power, let's say Mikrowatts, the calculation of the tick at which FlexOptions change will result in a very large value. The value might be so large that the `Long` value overflows, which we want to prevent. 

So there's only two things left to do from my side:
- check if the 1 W tolerance for a GWh installation suffices
- enhance the documentation of the value with a short version of the explanation (var overflow)